src/App.tsx

```diff
@@ -323,24 +319,22 @@ const App = (): JSX.Element => {
     };
   }, [dispatch, bridgeLoaded, address, wrappedTokenBalance, wrappedTokenTransferableBalance]);
 
+  const governanceTokenBalanceInvalidate = useGovernanceTokenBalanceInvalidate();
+
   // Subscribes to governance token balance
   React.useEffect(() => {
-    if (!dispatch) return;
     if (!bridgeLoaded) return;
     if (!address) return;
+    if (!governanceTokenBalanceInvalidate) return;
 
     (async () => {
       try {
         const unsubscribe = await window.bridge.tokens.subscribeToBalance(
           GOVERNANCE_TOKEN,
           address,
-          (_: string, balance: ChainBalance<GovernanceUnit>) => {
-            if (!balance.free.eq(governanceTokenBalance)) {
-              dispatch(updateGovernanceTokenBalanceAction(balance.free));
-            }
-            if (!balance.transferable.eq(governanceTokenTransferableBalance)) {
-              dispatch(updateGovernanceTokenTransferableBalanceAction(balance.transferable));
-            }
+          // TODO: it looks like the callback is called just before the balance is updated (not after)
+          () => {
```

Comment:
In case you want to simplify, you can just pass `governanceTokenBalanceInvalidate`. Im fine with both.